src/Raven.Client/Documents/Linq/RavenQueryInspector.cs

```diff
@@ -90,6 +91,18 @@ IEnumerator IEnumerable.GetEnumerator()
             return GetEnumerator();
         }
 
+        public async IAsyncEnumerator<T> GetAsyncEnumerator(CancellationToken cancellationToken = new CancellationToken())
+        {
+            var execute = GetAsyncDocumentQuery();
+            var list =  await ((IAsyncDocumentQuery<T>)execute)
+                .ToListAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
This will cause the query to be executed here (a request to the server will be sent). I'm don't think it is expected behavior.

```
var users = session.Query<User>().AsAsyncEnumerable(); <--- request will be sent here

await foreach (var user in users)
{
    ...
}
```

Reply:
Please check how do we deal with that in the query streaming

Reply:
We verify it in the test:
https://github.com/ravendb/ravendb/pull/15639/files#diff-38d5ea6680442fc0fcf6af9036267307f33681bdf28cf293de30fd4c158e4ecfR57

Reply:
Oh, I see now. When we call `session.Query<User>().AsAsyncEnumerable();` we don't go here to `GetAsyncEnumerator()` but we go just to the extension method:

https://github.com/ravendb/ravendb/pull/15639/files#diff-c9d804509c1093339f828ed6c26e337874484f5fbf63f0d94e34b255b7aa4fddR147-R148

So we don't evaluate that. Everything is good.